pvlib/shading.py

```diff
@@ -191,3 +192,40 @@ def sky_diffuse_passias(masking_angle):
        Available at https://www.nrel.gov/docs/fy18osti/67399.pdf
     """
     return 1 - cosd(masking_angle/2)**2
+
+
+def shaded_fraction(solar_zenith, solar_azimuth, surface_tilt, surface_azimuth,
+                    gcr):
```

Comment:
Eq 32 in https://www.nrel.gov/docs/fy20osti/76626.pdf is an extended version of this equation that also considers vertical offset between rows for arrays built on sloped terrain.  Worth adding an optional `cross_axis_slope=0` parameter here like the one in `tracking.singleaxis`?